### PR TITLE
Add manifest for ZAP

### PR DESCRIPTION
**Pull request by psiinon**
Signed-off-by: Simon Bennetts <psiinon@gmail.com>

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/winget-pkgs/pull/756)

### CODE REVIEW

**Comment by kingthorin** [comment]
@denelon could you let us know why the pipeline was unhappy when we'd included both x86 and x64?

**Comment by psiinon** [comment]
@kingthorin that was actually explained in one of the error messages: "Manifest: We don't support more than one installer for now." :)

**Comment by kingthorin** [comment]
Oh okay.

**Comment by denelon** [comment]
I got an "Installer failed with exit code: 83" in my test VM. Did you test this on your local machine with `winget install -m <manifest>`? Are there any library requirements?

**Comment by kingthorin** [comment]
Well it’s a Java app so needs Java available to run. Install-wise it uses install4j which I believe needs Java as well (and last I checked either Oracle or an Open variant that sets the Oracle reg keys). I don’t know if @psiinon had tested.

**Comment by psiinon** [comment]
I've managed to get winget installed from the repo. The manifest is failing for me too, even though I have java installed :/
It just says "Error opening media file." and refers to a log file that doesnt appear to exist :(
I'll try creating a ZAP release with the JRE included...

**Comment by KevinLaMS** [comment]
Removing passed label until the JRE fix is available.

**Comment by kingthorin** [comment]
@KevinLaMS is there a plan to allow packages to declare a dependency? So the ZAP package (or similar) could state that it requires a JRE? Or is there some other plan to address this?

**Comment by psiinon** [comment]
My plan was to try to generate a Windows installer with the JRE included .. but I got distracted by other things :/
Will try to get back to this asap...

**Comment by denelon** [comment]
I'm adding label for "Dependencies". The manifest schema also needs to be updated.

**Comment by kingthorin** [comment]
What author feedback is needed? Until dependency handling is sorted I don't see what we can do here.

**Comment by OfficialEsco** [comment]
Uhhh, ZAP 2.9.0 already got added in #1014 (@KevinLaMS 😠)
Submitted ZAP 2.10.0 with Dependency and correct information at #13654
Should we delete Zap.Zap until this PR gets approved under OWASP.ZAP?
https://github.com/microsoft/winget-cli/issues/163

**Comment by denelon** [comment]
@OfficialEsco If ZAP 2.9.0 requires dependencies, it would likely be best to remove it. What kind of dependency does it have?

The tags in this manifest should be converted to an array as well.

I also created this Issue to help #13861

**Comment by OfficialEsco** [comment]
@denelon they both require Java minimum version 1.8, the installer will not start if Java is not installed

**Comment by denelon** [comment]
@OfficialEsco thanks for the quick reply. I think this might be a quick win for dependencies. We aren't going to try and figure out which version of Java to install. It will likely be the "informational" type of dependency. A warning will be presented and the user could say [y]es to continue. 

https://github.com/microsoft/winget-cli/blob/master/doc/specs/%23163%20-%20Dependencies.md

**Comment by psiinon** [comment]
@denelon sounds good, we're v happy to trial this :) Is there an eta for this feature? How would it be best for us to track it?
We can update this PR for both the Java dependency and to pick up the latest version of ZAP...

**Comment by denelon** [comment]
@psiinon we have an engineer just about ready to start work on dependencies. Once we have the "informational" warning behavior, we may decide to review any of the manifests with dependencies to see if they can be included. There is still additional work to update the validation pipelines, so that would be the next step in the process.

**Comment by kingthorin** [comment]
@denelon any news?

**Comment by denelon** [comment]
We've [started working on dependencies](https://github.com/microsoft/winget-cli/pull/1552), but it's definitely a long tail feature. I'm expecting more progress on the experimental feature in our next release. It's likely going to take several iterations and refinement.

**Review by ItzLevvie — APPROVED** [review]

**Comment by kingthorin** [comment]
@ItzLevvie could you point us to some docs or an example where Java is a dependency?

**Comment by ItzLevvie** [comment]
> an example where Java is a dependency?

An example can be seen here (as well as its implementation): https://github.com/microsoft/winget-pkgs/pull/15785/files#diff-0d6ed1b64324bef7309b48244334cf17dbd9c0a016bb2343b55fc26d7b70a235R5-R7

Please do note that package dependencies probably won't be out until Q3 or Q4 of 2022; so we're still many months away from that which means adding that piece of snippet will still make the PR not pass through Azure Pipelines.

**Comment by kingthorin** [comment]
Okay, we’ll continue to standby. Please feel feee to update us as things get close.

**Comment by ItzLevvie** [comment]
[Policy] Needs-CLA

image

**Comment by ItzLevvie** [comment]
[Policy] reset feedback

**Review by ItzLevvie — APPROVED** [review]

**Comment by psiinon** [comment]
Is this likely to get merged?
Because we're now on ZAP v2.13.0 😛 
I can push an update, or can wait for it to be merged and then make a new PR .. whichever is most likely to not cause more delays...

**Comment by OfficialEsco** [comment]
I'd say make the PR up to date with the Dependency to see if everything goes trough as expected

**Comment by psiinon** [comment]
Updated to the latest version of ZAP

**Comment by OfficialEsco** [comment]
I'd recommend updating the manifest to schema version 1.4.0 since this one is ancient and not valid anymore. Singletons is also going away later.

You also need to include the Java dependency for it to pass.

**Comment by psiinon** [comment]
@OfficialEsco links would be appreciated. I submitted this over 3 years ago, its not surprising that things have changed, but getting back up to speed on it is also challenging...

**Comment by OfficialEsco** [comment]
Understandable =)
Here is the docs for the 1.4.0 schema with examples https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.4.0
Here are the tools to automatically create manifests https://github.com/microsoft/winget-pkgs#authoring-a-manifest

**Comment by psiinon** [comment]
FYI on holiday this week, but will aim to update this as soon as I can..

**Comment by OfficialEsco** [comment]
Quick Manifest:

<details>
<summary>OWASP.ZAP.installer.yaml</summary>

```yaml
# Created with YamlCreate.ps1 v2.2.8 $debug=NVS1.LF.7-3-4.Unix
# yaml-language-server: $schema=https://aka.ms/winget-manifest.installer.1.4.0.schema.json

PackageIdentifier: OWASP.ZAP
PackageVersion: 2.13.0
InstallerType: exe
InstallModes:
- silent
InstallerSwitches:
  Silent: -q
  SilentWithProgress: -q
UpgradeBehavior: install
ReleaseDate: 2023-07-12
Dependencies:
  PackageDependencies:
  - PackageIdentifier: Oracle.JavaRuntimeEnvironment
Installers:
- Architecture: x64
  InstallerUrl: https://github.com/zaproxy/zaproxy/releases/download/v2.13.0/ZAP_2_13_0_windows.exe
  InstallerSha256: 5E1DF341B91F65FE0768CB659AF41E7671D475074CEAB1B332B6520BCEDF1DAC
ManifestType: installer
ManifestVersion: 1.4.0

```

</details>

<details>
<summary>OWASP.ZAP.locale.en-us.yaml</summary>

```yaml
# Created with YamlCreate.ps1 v2.2.8 $debug=NVS1.LF.7-3-4.Unix
# yaml-language-server: $schema=https://aka.ms/winget-manifest.defaultLocale.1.4.0.schema.json

PackageIdentifier: OWASP.ZAP
PackageVersion: 2.13.0
PackageLocale: en-US
Publisher: OWASP
PublisherUrl: https://github.com/zaproxy/zaproxy
PublisherSupportUrl: https://github.com/zaproxy/zaproxy/issues
# PrivacyUrl:
Author: OWASP
PackageName: ZAP
PackageUrl: https://github.com/zaproxy/zaproxy
License: Apache License 2.0
LicenseUrl: https://raw.githubusercontent.com/zaproxy/zaproxy/main/LICENSE
# Copyright:
CopyrightUrl: https://raw.githubusercontent.com/zaproxy/zaproxy/main/LICENSE
ShortDescription: The worlds most frequently used web app scanner. Free and Open Source.
# Description:
# Moniker:
Tags:
- dast
- owasp
- scanner
- security
# ReleaseNotes:
ReleaseNotesUrl: https://www.zaproxy.org/docs/desktop/releases/2.13.0/
# PurchaseUrl:
# InstallationNotes:
# Documentations:
ManifestType: defaultLocale
ManifestVersion: 1.4.0

```

</details>

<details>
<summary>OWASP.ZAP.yaml</summary>

```yaml
# Created with YamlCreate.ps1 v2.2.8 $debug=NVS1.LF.7-3-4.Unix
# yaml-language-server: $schema=https://aka.ms/winget-manifest.version.1.4.0.schema.json

PackageIdentifier: OWASP.ZAP
PackageVersion: 2.13.0
DefaultLocale: en-US
ManifestType: version
ManifestVersion: 1.4.0

```

</details>

**Comment by stephengillie** [comment]
>  Manifest is invalid. Manifest file missing required field: PackageIdentifier

File: manifests/OWASP/ZAP/2.13.0.yaml
Comment (stephengillie) [inline manifests/OWASP/ZAP/2.13.0.yaml:1]:
```suggestion
PackageIdentifier: OWASP.ZAP
```

**Comment by ItzLevvie** [comment]
[Policy] Blocking-Issue

**Comment by psiinon** [comment]
@OfficialEsco ZAP is no longer part of OWASP. Are we ok just using "ZAP" as the identifier?

**Comment by psiinon** [comment]
Updated - hope thats ok 😁

**Comment by OfficialEsco** [comment]
@psiinon
```
Manifest: The filename must match the PackageIdentifier and ManifestType properties from the manifest.
actual 'ZAP.ZAP.locale.en-us.yaml'
expected 'ZAP.ZAP.locale.en-US.yaml'
```

**Comment by psiinon** [comment]
Lets see if that works..

**Comment by psiinon** [comment]
@OfficialEsco I cant see any evidence? Am I missing something?

**Comment by stephengillie** [comment]
Automatic Validation ended with:
> 2023-08-08 04:14:13.126 [CLI ] ShellExecute installer failed: 83 

image

**Comment by stephengillie** [comment]
Manual Validation failed with the same error. Manually installing failed with this error: 
image

I believe `Oracle.JavaRuntimeEnvironment` is only `Java 8` while `Java 11` is required here. Should a different JRE be used for the dependency?

**Comment by psiinon** [comment]
Yes, we need a min of Java 11.
Are there any v11 JREs available? If so what should I specify (or how can I find that out for myself).
Many thanks.

**Comment by OfficialEsco** [comment]
We only have the classic JRE + JDK from Oracle, what do you the developer recommend?
https://github.com/microsoft/winget-pkgs/tree/master/manifests/o/Oracle

We also have a bunch of other Java brands, we could use Microsoft's OpenJDK?
https://github.com/microsoft/winget-pkgs/tree/master/manifests/m/Microsoft/OpenJDK
All you have to do is add the `PackageIdentifier` of the package, so for example `Microsoft.OpenJDK.11`.
I'm not sure if the feature of using `Microsoft.OpenJDK` is implemented yet, it was supposed to pick any in that branch

**Comment by psiinon** [comment]
Trying `Microsoft.OpenJDK.11` 😁

**Comment by thc202** [comment]
It seems it's now failing because the installer requires admin privileges (exit code 0x8A150006). Is that supported? Do we need to specify something in the manifest?

File: manifests/z/ZAP/ZAP/2.13.0/ZAP.ZAP.installer.yaml
Comment (OfficialEsco) [inline manifests/z/ZAP/ZAP/2.13.0/ZAP.ZAP.installer.yaml:10]:
```suggestion
ReleaseDate: 2023-07-12
ElevationRequirement: elevationRequired
```


> ElevationRequirement - Indicator for elevation requirements when installing or upgrading packages.
>
> Optional Field
>
> This key represents which scope a package is required to be executed under. Some packages require user level execution while others require administrative level execution.
>
> * elevationRequired - Must be run from a shell that is running in an administrative context (e.g - Admin user using powershell/terminal/cmd with "Run as Administrator")
> * elevationProhibited - Must be run from a shell that is not running in an administrative context
> * elevatesSelf - If called from a non-administrative context, will request elevation. If called from an administrative context, may or may not request elevation.

**Comment by psiinon** [comment]
Thought it was worth a try 😉

File: manifests/z/ZAP/ZAP/2.13.0/ZAP.ZAP.installer.yaml
Comment (thc202) [reply manifests/z/ZAP/ZAP/2.13.0/ZAP.ZAP.installer.yaml:10]: Where are those docs?

Comment (thc202) [reply manifests/z/ZAP/ZAP/2.13.0/ZAP.ZAP.installer.yaml:10]: Found it https://github.com/microsoft/winget-pkgs/blob/master/doc/manifest/schema/1.4.0/installer.md :)

**Comment by OfficialEsco** [comment]
This is probably not relevant but
```shell
2023-08-09 03:26:36.097 [CLI ] Starting: 'C:\Users\VALIDA~1\AppData\Local\Temp\WinGet\ZAP.ZAP.2.13.0\ZAP_2_13_0_windows.exe' with arguments '-q'
2023-08-09 03:26:41.648 [CLI ] ShellExecute installer failed: 83
2023-08-09 03:26:41.649 [CLI ] Terminating context: 0x8a150006 at D:\a\_work\1\s\external\pkg\src\AppInstallerCLICore\Workflows\InstallFlow.cpp:1bf
```

**Comment by thc202** [comment]
Maybe it is:
```
The process of an installer returns 0 if the installation was completed successfully, 1 if the installation fails and 83 if the installer could not find a suitable JVM to run.
```
https://www.ej-technologies.com/resources/install4j/help/doc/installers/errors.html

I thought this was the actual problem:
```
Installation failed with exit code -1978335226
```

**Comment by OfficialEsco** [comment]
I'm unsure if Dependencies are enabled in Validation as its in experimental, @stephengillie should be able to answer that in a few hours

**Comment by denelon** [comment]
We're getting close to dependencies shipping in WinGet 1.6.
After the client starts to roll out and gets >50% market penetration (~ 2 weeks after release) we will open up dependencies in all manifests. We're currently grandfathering packages that had the field well ahead of "full" support while we were working on the feature.

**Comment by psiinon** [comment]
Thanks @denelon - what do you recommend we do right now? Is there a workaround?

**Comment by denelon** [comment]
I'd love to say I have the "right" answer here, but it's a bit tricky. We don't really know if the user has dependencies enabled in their client to have a subjective "install notes" that would inform them about the need to enable dependencies, and if the package install fails without having Java installed, we don't get a good experience.

I'd say it's a question of what percentage of people installing this package would not have Java installed, and how popular is the package? 

Responding to complaints that it failed to install is time consuming and takes away from product development. 

@OfficialEsco what are your thoughts here as a long-time contributor/moderator?

**Comment by Trenly** [comment]
> If there is a single return response for missing dependency, then I'd be OK adding that to the manifest and adding the dependency and doing a manual approval.
> 
> @Trenly I think this pattern might make sense for some of the other packages with dependencies until we get validation working with the dependency install flow logic.

I think that makes sense. However, I don't know how common it is to have a return code specifically for missing dependency as opposed to other installation errors. Even looking at the [MSI documentation](https://learn.microsoft.com/en-us/windows/win32/msi/error-codes) I don't see a code specific for missing dependency - only 1603 for installation failure

File: manifests/z/ZAP/ZAP/2.13.0/ZAP.ZAP.installer.yaml
Comment (OfficialEsco) [inline manifests/z/ZAP/ZAP/2.13.0/ZAP.ZAP.installer.yaml:11]:
```suggestion
ElevationRequirement: elevationRequired
ExpectedReturnCodes:
  - InstallerReturnCode: 83
    ReturnResponse: missingDependency
    ReturnResponseUrl: https://www.ej-technologies.com/resources/install4j/help/doc/installers/errors.html
```

I guess this is how we can do it since 83 is the missing dependency exit code
Also linking https://github.com/microsoft/winget-cli/issues/360 since the InstallerType is Install4j

**Comment by psiinon** [comment]
Have applied that suggestion...

**Comment by denelon** [comment]
@psiinon if you enable the dependencies feature in WinGet and install this package via a local manifest on a clean machine, does it correctly install the Java requirement and the package?

**Comment by Trenly** [comment]
> PS D:\Git\winget-pkgs> .\Tools\PRTest.ps1 756 -EnableExperimentalFeatures -WinGetVersion 1.6 -Prerelease

After agreeing to msstore agreements - it installs normally
```
--> Installing WinGet
--> Disabling safety warning when running installer
Tip: you can type 'Update-EnvironmentVariables' to update your environment variables, such as after installing a new software.

--> Configuring Winget
Enabled admin setting 'LocalManifestFiles'.
Enabled admin setting 'LocalArchiveMalwareScanOverride'.


--> Installing the Manifest 2.13.0

Found ZAP [ZAP.ZAP] Version 2.13.0
This application is licensed to you by its owner.
Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
This package requires the following dependencies:
  - Packages
      EclipseAdoptium.Temurin.11.JRE
The `msstore` source requires that you view the following agreements before using.
Terms of Transaction: https://aka.ms/microsoft-store-terms-of-transaction
The source requires the current machine's 2-letter geographic region to be sent to the backend service to function properly (ex. "US").

Do you agree to all the source agreements terms?
[Y] Yes  [N] No: y
Installing dependencies:
(1/1) Found Eclipse Temurin JRE with Hotspot 11 [EclipseAdoptium.Temurin.11.JRE] Version 11.0.19.7
This application is licensed to you by its owner.
Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
Downloading https://github.com/adoptium/temurin11-binaries/releases/download/jdk-11.0.19+7/OpenJDK11U-jdk_x64_windows_hotspot_11.0.19_7.msi
  ██████████████████████████████   166 MB /  166 MB
Successfully verified installer hash
Starting package install...
Successfully installed

Downloading https://github.com/zaproxy/zaproxy/releases/download/v2.13.0/ZAP_2_13_0_windows.exe
  ██████████████████████████████   194 MB /  194 MB
Successfully verified installer hash
Starting package install...
Successfully installed

--> Refreshing environment variables

--> Comparing ARP Entries

DisplayName                                      DisplayVersion Publisher        ProductCode
-----------                                      -------------- ---------        -----------
OWASP Zed Attack Proxy 2.13.0                    2.13.0         OWASP ZAP        OWASP ZAP
Eclipse Temurin JDK with Hotspot 11.0.19+7 (x64) 11.0.19.7      Eclipse Adoptium {9A277E09-3F69-4362-8074-A49E765350B0}
```

File: manifests/z/ZAP/ZAP/2.13.0/ZAP.ZAP.installer.yaml
Comment (Trenly) [inline manifests/z/ZAP/ZAP/2.13.0/ZAP.ZAP.installer.yaml:1]:
```suggestion
# yaml-language-server: $schema=https://aka.ms/winget-manifest.installer.1.4.0.schema.json

PackageIdentifier: ZAP.ZAP
```

File: manifests/z/ZAP/ZAP/2.13.0/ZAP.ZAP.yaml
Comment (Trenly) [inline manifests/z/ZAP/ZAP/2.13.0/ZAP.ZAP.yaml:1]:
```suggestion
# yaml-language-server: $schema=https://aka.ms/winget-manifest.version.1.4.0.schema.json

PackageIdentifier: ZAP.ZAP
```

**Review by Trenly — COMMENTED** [review]
Including the schema headers is best for intellisense and anyone using VSCode

**Comment by denelon** [comment]
OK, since we're so close to getting dependencies in, and this PR has had so much activity, I'm inclined to approve. 

@Stephen Gillie if you've gotten past all our mandatory policy checks, this one is OK to go ahead and merge. If you haven't, reach out to me so we can find a way to verify everything.

File: manifests/z/ZAP/ZAP/2.13.0/ZAP.ZAP.locale.en-US.yaml
Comment (Trenly) [inline manifests/z/ZAP/ZAP/2.13.0/ZAP.ZAP.locale.en-US.yaml:4]:
```suggestion
Publisher: OWASP ZAP
```
This should match control panel

**Comment by denelon** [comment]
I spoke too soon. @Trenly has some changes requested.

File: manifests/z/ZAP/ZAP/2.13.0/ZAP.ZAP.locale.en-US.yaml
Comment (Trenly) [inline manifests/z/ZAP/ZAP/2.13.0/ZAP.ZAP.locale.en-US.yaml:9]:
```suggestion
PackageName: OWASP Zed Attack Proxy
```
This should match Control Panel as closely as possible

**Review by Trenly — CHANGES_REQUESTED** [review]

**Comment by Trenly** [comment]
> I spoke too soon. @Trenly has some changes requested.

Sorry to spoil your plans 😢 
Once changes are accepted, I think it should be good to merge

File: manifests/z/ZAP/ZAP/2.13.0/ZAP.ZAP.locale.en-US.yaml
Comment (psiinon) [reply manifests/z/ZAP/ZAP/2.13.0/ZAP.ZAP.locale.en-US.yaml:4]: Is this really necessary? 
ZAP is no longer part of OWASP. We will be releasing a new version, but dont have a date for it yet.

File: manifests/z/ZAP/ZAP/2.13.0/ZAP.ZAP.locale.en-US.yaml
Comment (psiinon) [reply manifests/z/ZAP/ZAP/2.13.0/ZAP.ZAP.locale.en-US.yaml:9]: Fine for it to be "Zed Attack Proxy", but would definitely prefer it not to include "OWASP"

Comment (OfficialEsco) [reply manifests/z/ZAP/ZAP/2.13.0/ZAP.ZAP.locale.en-US.yaml:9]: This metadata is used to match previous and future versions to this Package identifier.

Comment (OfficialEsco) [reply manifests/z/ZAP/ZAP/2.13.0/ZAP.ZAP.locale.en-US.yaml:11]: Well, don't you want people to update from this version to the next? if so the PackageName should be updated, or.. you could cheese it by adding the ProductCode,  but then please update the Publisher and PackageName to what it actually will display in the control panel

Comment (psiinon) [reply manifests/z/ZAP/ZAP/2.13.0/ZAP.ZAP.locale.en-US.yaml:11]: Yes, we want people to be able to update from this version to the next, which is why we dont want to include "OWASP" in the Publisher or Package Name 😁 
If thats a requirement then we'll put this PR on hold until we have a new version...

File: manifests/z/ZAP/ZAP/2.13.0/ZAP.ZAP.locale.en-US.yaml
Comment (Trenly) [reply manifests/z/ZAP/ZAP/2.13.0/ZAP.ZAP.locale.en-US.yaml:6]: If you want to leave (and the Package Name) this as just `ZAP` you would need to add the `AppsAndFeaturesEntries` section in the Installer manifest so that winget can detect the version that is installed. However, if you set AppsAndFeaturesEntries for one version, it will also need to be set for all versions going forward (although the values will change to match what is written to the registry)

This would be -
```yaml
AppsAndFeaturesEntries:
- DisplayName: OWASP Zed Attack Proxy 2.13.0
  Publisher: OWASP ZAP
  ProductCode: OWASP ZAP
```

File: manifests/z/ZAP/ZAP/2.13.0/ZAP.ZAP.locale.en-US.yaml
Comment (psiinon) [reply manifests/z/ZAP/ZAP/2.13.0/ZAP.ZAP.locale.en-US.yaml:11]: So we can have "OWASP" in the Publisher and Package name for this release but then change it for the next release (which will match the Control Panel) without that causing any problems?

Comment (Trenly) [reply manifests/z/ZAP/ZAP/2.13.0/ZAP.ZAP.locale.en-US.yaml:11]: Yes, that should work, assuming the control panel entries would be changed in the next release to also be without `OWASP`

File: manifests/z/ZAP/ZAP/2.13.0/ZAP.ZAP.locale.en-US.yaml
Comment (mdanish-kh) [reply manifests/z/ZAP/ZAP/2.13.0/ZAP.ZAP.locale.en-US.yaml:6]: > if you set AppsAndFeaturesEntries for one version, it will also need to be set for all versions going forward

IIRC, that's only if we have DisplayVersion as it causes additional version mapping. I don't think the other ARP entries are an issue

Comment (Trenly) [reply manifests/z/ZAP/ZAP/2.13.0/ZAP.ZAP.locale.en-US.yaml:6]: > IIRC, that's only if we have DisplayVersion as it causes additional version mapping. I don't think the other ARP entries are an issue

Hmm, good point. I think you might be right, but still not a bad thing to carry it forward

**Comment by psiinon** [comment]
OK, have applied those 2 suggestions...

**Comment by stephengillie** [comment]
Automatic Validation ended with:
> Installation failed with exit code -1978334972 
2023-08-16 00:36:22.254 [CLI ] ShellExecute installer failed: 83 
image

**Comment by stephengillie** [comment]
Manual Validation couldn't replicate the error. The package installed and launched normally.

**Review by stephengillie — APPROVED** [review]

**Comment by psiinon** [comment]
Woohoo!
Thanks everyone for your assistance getting this merged!

**Pull request merged**
